### PR TITLE
Add cell tracking example

### PR DESCRIPTION
**Pull request by Olusesan**
Issue
The codebase mostly includes synthetic data for learning and demonstration purposes. This limits the ability to:
- Demonstrate authentic cell morphology and behavior patterns
- Test edge cases that occur in actual experimental conditions
- Provide users with realistic examples for learning

Fix
This PR adds a curated dataset of real cell culture images to enhance the viewer's testing capabilities and user experience. Added real cell culture dataset (/examples/cell_tracking.py/) containing:

- High-resolution phase contrast microscopy images
- Time-lapse sequences showing cell division and migration

### CODE REVIEW

**Comment by TimMonko** [comment]
Hi @Olusesan!
So glad to see this example! Here's just a quick driveby:

First thing I found running locally is that this needs `imagecodecs` installed. We have a `gallery` optional-dependency  that you could add `tifffile[imagecodecs]` to. 

Second, the description about the example suggests that this should also have tracking data. As is, I'm only getting the addition of the downloaded 'Cell Tracking' image layer, but no tracks layer itself. What is your goal with this example? Showing how to load a remote dataset, or also including a tracks layer with properties (as description says) 

Finally, the folder does not clean itself up after install, so it leaves a `napari-cell-tracking` in my AppData\Local directory, which I'd generally consider less ideal, and a protected (hidden) part of a Windows environment. If it cleaned itself up after I closed the viewer, then that would be a-ok with me.

**Comment by psobolewskiPhD** [comment]
Yeah I agree with Tim here, it feels like the example is missing the meat and potatoes: the actual tracking part.

Regarding the files, I think it's the expected behavior of pooch to locally cache the files. The location is determined by platform:
https://www.fatiando.org/pooch/latest/api/generated/pooch.os_cache.html#pooch-os-cache

It's basically handled like the sample data we use for napari.
Look for a scikit-image and napari dir in the same place, ie. C:\Users\<user>\AppData\Local\

We could use `napari` as the location instead of a the more specific one. I already have a pile of stuff in Caches/napari

File: examples/cell_tracking.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    download(f"doi:{doi}", output_file=data_path, pooch=None)
+else:
+    print(f"using cached {data_path}")
+
+with zipfile.ZipFile(data_path, 'r') as zip_ref:
+    zip_ref.extractall(tmp_dir)
+
+print("Contents of tmp_dir after extraction:")
+for p in tmp_dir.rglob("*"):
+    print(p)
```
Comment (psobolewskiPhD) [inline examples/cell_tracking.py]: On my end this was quite spammy, maybe instead we can just print `tmp_dir` so the user knows where the files were saved/extracted? 
something like
`f"Extracted {n} files to {tmp_dir}"`

File: examples/cell_tracking.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+#imgs = glob.glob(os.path.join(tmp_dir, "*.tif"))
+imgs = list((tmp_dir/"01").glob ("*.tif"))
+
+if not imgs:
+    raise FileNotFoundError("no tif files found")
+
+images = np.stack([
+        tifffile.imread(fn) for fn in sorted(imgs)
```
Comment (psobolewskiPhD) [inline examples/cell_tracking.py]: This requires imagecodecs, so please add:
`tifffile[codecs]`
to:
https://github.com/napari/napari/blob/f536ffeafa703ad8d2eb6183d9b9fcb871c326a8/pyproject.toml#L167

File: examples/cell_tracking.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+url = "https://zenodo.org/records/15832699/files/enhanced_gnn_tracks.csv?download=1"
+data_dir = pooch.retrieve(
+    url=url,
+    known_hash="sha256:a410154f3d0cdd2807f0aa66a87527c8125816979cd035fc9b61378551fe7b52",
+    fname="enhanced_gnn_tracks",
+    path=pooch.os_cache("my_tracking_project")
+)
+
+tracks_data = load_tracks_data(Path(data_dir).parent, filename=Path(data_dir).name)
```
Comment (psobolewskiPhD) [inline examples/cell_tracking.py]: I think it would be cleaner to do this part at the top, along with the tiffs?
So have a section just for downloading stuff, the tiffs and the csv

File: examples/cell_tracking.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        color_by='is_splitting',
+        blending='additive'
+    )
+# Add cell lebels
+label_url = 'https://zenodo.org/records/15852284/files/masks_pred.npz?download=1'
+label_path = pooch.retrieve(url=label_url,
+                            fname="masks_pred.npz",
+                            known_hash=None,
+                            path=pooch.os_cache("cell_labels")
+                            )
```
Comment (psobolewskiPhD) [inline examples/cell_tracking.py]: Same as above, I think it would be better to do all the downloading and file i/o at the top of the file, together.

File: examples/cell_tracking.py (hunk 2 above)
Comment (psobolewskiPhD) [reply examples/cell_tracking.py]: This ~~will~~ should fix the CI failures with tests and docs.

File: examples/cell_tracking.py (hunk 1 above)
Comment (Olusesan) [reply examples/cell_tracking.py]: Copy that, I will add a formatted string to print out the directory for saved files.

File: examples/cell_tracking.py (hunk 2 above)
Comment (Olusesan) [reply examples/cell_tracking.py]: Thanks for catching this, the toml file shall be updated accordingly.

File: examples/cell_tracking.py (hunk 3 above)
Comment (Olusesan) [reply examples/cell_tracking.py]: That makes sense.. the script is a sequence of attempts at solving different tasks as encountered which is why it looks this way. All downloads are now going to be moved up to make it look cleaner.

File: examples/cell_tracking.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
+"""
+Updated Cell tracking example
+==============================
```
Comment (psobolewskiPhD) [inline examples/cell_tracking.py]:
```suggestion
Cell tracking
=============
```
"Updated" doesn't mean much to anyone else and this is our examples gallery, so I think example is redundant.

BTW It's critical that the number of = match the number of chars in the title. Right now docs build is failing because of the extra =

File: examples/cell_tracking.py [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+This example demonstrates how the tracking layer is used
+for displaying 2D + time dataset of cells with colored properties for track-id.
```
Comment (psobolewskiPhD) [inline examples/cell_tracking.py]:
```suggestion
This example demonstrates how the Track layer can be used for
visualizing cell tracking data, by displaying a 2D + time dataset of 
cells with colored properties for track-id.
```
Maybe something more like this?

File: examples/cell_tracking.py [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+import numpy as np
+import pandas as pd
+import pooch
+import tifffile
+
+import napari
+
+###############################################################################
+# Download data
+# ------------------
```
Comment (psobolewskiPhD) [inline examples/cell_tracking.py]: I would either remove this heading or add additional ones for other sections of the example. As is it's a bit odd to me?

File: examples/cell_tracking.py (hunk 5 above)
Comment (Olusesan) [reply examples/cell_tracking.py]: Got it, edited "=" to match chars.

File: examples/cell_tracking.py (hunk 7 above)
Comment (Olusesan) [reply examples/cell_tracking.py]: All of the downloads have been moved to the top and I copied this header style from other examples in the repo. So maybe keep it for consistency?

Comment (psobolewskiPhD) [reply examples/cell_tracking.py]: But if there are no other sections, then I think this section divider seems out of place?
I dunno, I will defer to what others think.

File: examples/cell_tracking.py [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+
+This example demonstrates how the track layer is used
+for visualizing cell tracking data, by displaying 2D + time dataset 
+of cells with colored properties for track-id.
+
+Thanks to Dr. Alessia Ruggieri and Philipp Klein, Centre for Integrative Infectious
+Disease Research (CIID), University Hospital Heidelberg, Germany for the data.
+You can find the data on: https://doi.org/10.5281/zenodo.15597019
+
+"""
```
Comment (psobolewskiPhD) [inline examples/cell_tracking.py:14]:
```suggestion
.. tags:: visualization-advanced, layers
"""
```
Derp sorry I missed this before! You need to add some tags to the example so they show up properly in the gallery. This is making docs fail right now.
These are my suggestions, you can see existing tags here:
https://napari.org/stable/_tags/tagsindex.html

You can also add your own!

Comment (Olusesan) [reply examples/cell_tracking.py:14]: No worries man, this is the first time I am hearing of the tags in docs and thanks for sharing. I wouldn't have guessed they were the culprit here.

File: examples/cell_tracking.py (hunk 7 above)
Comment (TimMonko) [reply examples/cell_tracking.py]: I agree the divider seems out of place here, but I will leave it up to Sesan

**Comment by psobolewskiPhD** [comment]
Docs are building correctly now!
Here's the preview:
https://output.circle-artifacts.com/output/job/b630f6de-4541-41e1-8c1c-7b78f7d75cc8/artifacts/0/docs/docs/_build/html/gallery/cell_tracking.html#sphx-glr-gallery-cell-tracking-py
Here's what I mean about the Download header being the only one:
image

See how there is a short code block of setup and then a huge code block under Downloading?
It makes me thing everything is related to downloading.

File: examples/cell_tracking.py (hunk 7 above)
Comment (Olusesan) [reply examples/cell_tracking.py]: Cool, divider has been removed as requested.

File: examples/cell_tracking.py [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+                            fname="masks_pred.npz",
+                            known_hash=None,
+                            path=pooch.os_cache("cell_labels")
+                            )
+
+label_data = np.load(label_path)
+labels = label_data["masks"]
+
+viewer.add_labels(labels, name="Predicted Masks")
+napari.run()
```
Comment (psobolewskiPhD) [inline examples/cell_tracking.py]:
```suggestion

if __name__ == '__main__':
    napari.run()
```

Gah should have caught this earlier, napari.run() is only needed for scripts, so we want to ensure things work for people using jupyter or ipython. See: https://napari.org/stable/developers/contributing/documentation/index.html#adding-examples-to-the-gallery

Comment (Olusesan) [reply examples/cell_tracking.py]: Thanks! I added the main block and I hope this fixes the issues.

**Comment by TimMonko** [comment]
Going to bed, but check out these warnings for more info on things that will need to be changed https://results.pre-commit.ci/run/github/144513571/1754451303.PdJtesT-RSKhWaqV47xAVg
We can help if needed!

**Comment by Olusesan** [comment]
> Going to bed, but check out these warnings for more info on things that will need to be changed https://results.pre-commit.ci/run/github/144513571/1754451303.PdJtesT-RSKhWaqV47xAVg We can help if needed!

Thanks Tim, I've fixed the highlighted lint issues with the new update.

**Comment by brisvag** [comment]
This is milestoned for 0.6.5; I haven;'t followed the long discussion so I'm not sure what's the status. @psobolewskiPhD you seem to be involved the most, what do you recommend here?

**Comment by TimMonko** [comment]
@Olusesan still has a few things they are working on for this example, including how to properly segment / share the data. For now, based on Lorenzo's recent efforts, I'm going to remove it from the milestone. Sesan, please continue to reach out, join community meetings, and work on this in the time that you have! We would love to include this in the gallery once it is ready.

**Comment by Olusesan** [comment]
Thanks for the follow up @TimMonko and sorry for the delay @brisvag. The past two weeks has been very chaotic personally hence the delay but chaos is getting back to baseline and I should be able to complete this soon.

**Comment by brisvag** [comment]
No pressure @Olusesan, thanks for checking back! We can get this in for the next release :)

File: examples/cell_tracking.py [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+tif_url = "https://zenodo.org/records/17267522/files/01.tar.gz?download=1"
+tif_path = pooch.retrieve(
+    url=tif_url,
+    fname="01.tar.gz",
+    known_hash=None,
+    path=pooch.os_cache("napari_cell_tracking_example")
+)
+
+print(f"loading predicted masks from {st_path}")
+print(f"Loading raw images from {tif_path}")
```
Comment (TimMonko) [inline examples/cell_tracking.py]: nit: these could be comments in the appropriate places

File: examples/cell_tracking.py [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+tif_folder = tmp_dir/ "01"
+
+# Extract raw images
+with tarfile.open(tif_path, 'r:gz') as tar:
+    clean_members = []
+    for member in tar.getmembers():
+        path_parts = Path(member.name).parts
+        # Skip cached apple double compression files (for mac users)
+        if not any(part.startswith('._') or part == '__MACOSX' for part in path_parts):
+            clean_members.append(member)
```
Comment (jni) [inline examples/cell_tracking.py]: Two comments from the 2025-11-13/14 Pacific community meeting:

- Juan: we can probably clean up the zip so that we don't have to filter so much. Will check flags to the `zip` utility to see if we can exclude these from the get-go.
- Tim: Also, pooch might have a built-in option to unzip and return a file iterator.

File: examples/cell_tracking.py [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+    print(f"Extracting {gt_path} to {gt_folder}")
+    with zipfile.ZipFile(gt_path, 'r') as zip_ref:
+        zip_ref.extractall(gt_folder)
+else:
+    print(f"Using cached ground truth at {gt_folder}")
+
+# Show content of extractions for validation/dubugging
+print("Contents of tmp_dir after extraction:")
+for p in tmp_dir.rglob("*"):
+    print(f"Extracted {p} files to {tmp_dir}")
```
Comment (TimMonko) [inline examples/cell_tracking.py]: nit: would also remove this for the final form

File: examples/cell_tracking.py (hunk 10 above)
Comment (Olusesan) [reply examples/cell_tracking.py]: Hi Tim - do you mean getting rid of the print statements?

File: examples/cell_tracking.py (hunk 11 above)
Comment (Olusesan) [reply examples/cell_tracking.py]: Hi Juan - I am using a new cleaned up widows zip file and I will get rid of the double compression cleanup code. This looks like the best way forward.

File: examples/cell_tracking.py (hunk 10 above)
Comment (TimMonko) [reply examples/cell_tracking.py]: Yep, I would remove the print statements. But, perhaps you like this information from a documentation and organizational perspective, so in that case I would make a comment in the appropriate line, like `# load predicted masks` if you want that type of documentation

**Comment by psobolewskiPhD** [comment]
The docs build fails (and example test fails) are due to missing dependency:
```
Extension error (sphinx_gallery.gen_gallery):
Handler <function generate_gallery_rst at 0x7fbe8de8a2a0> for event 'builder-inited' threw an exception (exception: No module named 'trackastra')
```
You'll need to update pyproject.toml to include `trackastra` in the `gallery` dependencies.

File: examples/cell_tracking.py [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+with zipfile.ZipFile(tif_path, 'r') as zip:
+    zip.extractall(path=tif_folder)
+
+# Sort through images and stack them
+imgs = [f for f in sorted((tif_folder).rglob ("*.tif"))
+        if not f.name.startswith("._")]
+
+valid_files = []
+
+for f in imgs:
```
Comment (DragaDoncila) [inline examples/cell_tracking.py]: Is this block of code just checking that the files are valid tiffs before loading them? If so, I think we could remove this and rely on the downloaded data containing valid TIFFs. My goal is to remove as much code as possible that doesn't actually focus on the track loading and display.

File: examples/cell_tracking.py [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+    except Exception as e:
+        print(f"Skipping {f.name} — not a valid TIFF ({e})")
+
+
+images = np.stack([
+        tifffile.imread(fn) for fn in sorted(imgs)
+        ])
+print(f"image shape is {images.shape}")
+
+# Make sure mask shape and image shape is the same
```
Comment (DragaDoncila) [inline examples/cell_tracking.py]: Same here, I would remove the checks on the format and length of the masks, and just load them in a single line.

File: examples/cell_tracking.py (hunk 9 above)
Comment (DragaDoncila) [reply examples/cell_tracking.py]: @Olusesan I don't see the main block here. Perhaps you missed pushing it?

File: examples/cell_tracking.py [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+# Quick check to see if everything looks ok
+print(f"DataFrame columns: {tracks_df.columns.tolist()}")
+print(f"DataFrame shape: {tracks_df.shape}")
+print(f"First few rows:\n{tracks_df.head()}")
+
+# Add Napari viewer
+viewer = napari.Viewer()
+
+# I am not sure if we should also open ground truth data here
+#viewer.open(str(gt_folder), plugin='napari-ctc')
```
Comment (DragaDoncila) [inline examples/cell_tracking.py]: I think probably let's not add ground truth. I notice we don't ever add the images themselves though, and that would be nice.

Comment (DragaDoncila) [reply examples/cell_tracking.py]: In fact maybe we can load the images directly with `images = viewer.open(str(tif_folder), plugin='napari-ctc')` and then we wouldn't have to load the tiffs individually ourselves above?

File: examples/cell_tracking.py [hunk 16]
```diff
@@ -0,0 +1,132 @@
...
+from trackastra.tracking import ctc_to_napari_tracks, graph_to_ctc
+
+import napari
+
+# Create temporary directory
+tmp_dir = Path(pooch.os_cache('napari-cell-tracking-example'))
+tmp_dir.mkdir(parents=True, exist_ok=True)
+
+#Download ground truth data
+gt_url = 'https://zenodo.org/records/16884151/files/TRA.zip?download=1'
```
Comment (DragaDoncila) [inline examples/cell_tracking.py]: If we don't open the ground truth data at all, perhaps we can also skip downloading it.

**Review by DragaDoncila — COMMENTED** [review]
@Olusesan this is getting very close 🤩 Excited to have it in. 

I left a few comments, mainly suggesting removing code where we can, to streamline the example and bring more attention to the actual tracking bit.

File: examples/cell_tracking.py (hunk 13 above)
Comment (TimMonko) [reply examples/cell_tracking.py]: Yes, for reference I cleaned up the Zip and it now only contains valid tiffs, so there does not need to be any file checking

File: examples/cell_tracking.py (hunk 15 above)
Comment (Olusesan) [reply examples/cell_tracking.py]: Thanks for clarifying the ground truth data! I'm not sure what you mean by main block and I also don't have reference for line 171, latest file commit stops at line 133.

Comment (DragaDoncila) [reply examples/cell_tracking.py]: @Olusesan yeah the line numbers got a bit messy. If you go to the Files Changed tab on GitHub and look at line 132, you'll see the context for my comment (screenshot below): 

image

Comment (Olusesan) [reply examples/cell_tracking.py]: Oh I see. Will add the main block back in. Thanks.

File: examples/cell_tracking.py [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+tmp_dir.mkdir(parents=True, exist_ok=True)
+
+# Extract silver truth data for trackastra
+st_url = "https://zenodo.org/records/15852284/files/masks_pred.npz?download=1"
+st_path = pooch.retrieve(
+    url=st_url,
+    fname="masks_pred.npz",
+    known_hash=None,
+    path=pooch.os_cache("napari cell tracking example")
+)
```
Comment (TimMonko) [inline examples/cell_tracking.py:36]: I can confirm you can do the unzipping in the retrieve stage with `processor=pooch.Unzip()`. So you can clean up the code additionally by switching to this logic, which will return a list of the files.
https://www.fatiando.org/pooch/latest/api/generated/pooch.Unzip.html#pooch.Unzip

```suggestion
st_path = pooch.retrieve(
    url=st_url,
    fname="masks_pred.npz",
    known_hash=None,
    path=pooch.os_cache("napari cell tracking example"),
    processor=pooch.Unzip(),
    progressbar=True,
)
```

Comment (Olusesan) [reply examples/cell_tracking.py:36]: Thanks Tim! The extraction is actually for the tif files and not masks. It's cleaner, now down to 75 lines and I like it. New commit is incoming.

**Comment by TimMonko** [comment]
@napari-bot update constraints

**Comment by TimMonko** [comment]
@napari-bot update constraints

**Comment by TimMonko** [comment]
@brisvag and @Czaki, I'm a bit concerned with how much the bot updated the constraints. Can you explain what's wrong? The docs/gallery are still failing because its not finding the new deps (namely, `trackastra`)

**Comment by psobolewskiPhD** [comment]
Isn't it because the dependency groups used by uv weren't updated? just the optional dependencies?
See: https://github.com/Olusesan/napari/blob/6bd8754ae35ff58d13625d37ddbc8abce22b56f2/pyproject.toml#L234

(also I expect CI to fail with the automated constraints update, because of the llvm/numba issue for docs and the pyside6 issue, see: https://github.com/napari/napari/pull/8456 )

Edit: otherwise I think the bot did the right thing: our constraints are out of date (due to the issue mentioned above) and trackastra adds a buncha new things.

File: examples/cell_tracking.py [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+    path=pooch.os_cache("napari_cell_tracking_example"),
+    processor=pooch.Unzip(),
+)
+
+# Load the downloaded masks
+masks_npz = np.load(st_path)
+masks = masks_npz['masks']
+
+# Sort through images and stack them
+imgs = sorted(tif_file.rglob("*.tif"))
```
Comment (DragaDoncila) [inline examples/cell_tracking.py]: `tif_file` is just a list of strings here, so the `rglob` doesn't work. We can just use vanilla string sorting though, because the naming convention works with just alphabetical sort.

```suggestion
imgs = sorted(tif_file)
```

File: examples/cell_tracking.py [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+
+# Generate tracks
+graph, *_ = model.track(imgs=images, masks=masks, mode='greedy')
+tracks_df, tracked_masks = graph_to_ctc(graph=graph, masks_original=masks)
+
+napari_tracks, napari_graph = ctc_to_napari_tracks(segmentation=tracked_masks, man_track=tracks_df)
+
+# Add Napari viewer
+viewer = napari.Viewer()
+viewer.add_labels(masks, name='Predicted Masks')
```
Comment (DragaDoncila) [inline examples/cell_tracking.py:71]: Let's add the raw images of the cells as well, so folks know what the underlying data was!

```suggestion
viewer.add_image(images, name='Images')
viewer.add_labels(masks, name='Predicted Masks')
```

**Review by DragaDoncila — APPROVED** [review]
Aside from the constraints issues, this example will work as expected with my proposed changes below. I'm going to go ahead and commit them. Thanks for all your work on this @Olusesan 😍

**Comment by TimMonko** [comment]
> Isn't it because the dependency groups used by uv weren't updated? just the optional dependencies? See: [Olusesan/napari@`6bd8754`/pyproject.toml#L234](https://github.com/Olusesan/napari/blob/6bd8754ae35ff58d13625d37ddbc8abce22b56f2/pyproject.toml#L234)
> 
> (also I expect CI to fail with the automated constraints update, because of the llvm/numba issue for docs and the pyside6 issue, see: #8456 )
> 
> Edit: otherwise I think the bot did the right thing: our constraints are out of date (due to the issue mentioned above) and trackastra adds a buncha new things.

I haven't been following enough, I thought we had merged #8456 . Is this why the uv constraints weren't updated? or is that an issue with using napari-bot? I'm so tired sorry 🤮

Wait I think I follow. But isn't pre-commit supposed to yell at me if they are out of sync ? it did yell at me because I didn't have py3.11, but maybe it was bypassed somehow by this PR originally (or the PR is old enough to be before the pre-commit addition)

Added to dependency groups now

**Comment by TimMonko** [comment]
@Olusesan would you like to be added to the napari citation file? If so please reply below filling this out, or adding your entry in alphabetical order to CITATION.cff and updating your PR:

```
- given-names: <first name>
  family-names: <last name>
  alias: Olusesan
  affiliation: <required>
  orcid: <optional>
```

**Comment by melonora** [comment]
Great work @Olusesan ! And was nice meeting you in person in SF!

**Comment by Olusesan** [comment]
> Aside from the constraints issues, this example will work as expected with my proposed changes below. I'm going to go ahead and commit them. Thanks for all your work on this @Olusesan 😍

Thanks @DragaDoncila! Really appreciate all the support and nice to see it approved

**Comment by Olusesan** [comment]
> @Olusesan would you like to be added to the napari citation file? If so please reply below filling this out, or adding your entry in alphabetical order to CITATION.cff and updating your PR:
> 
> ```
> - given-names: <first name>
>   family-names: <last name>
>   alias: Olusesan
>   affiliation: <required>
>   orcid: <optional>
> ```

Thanks @TimMonko and yes, it would be nice to be added to the citation file. Below is the completed requirements in the same order:

>   given-names: Sesan
>   family-names: Ajina
>   alias: Olusesan
>   affiliation: iLABS, Inc
>   orcid: 0000-0003-2910-1190

**Comment by Olusesan** [comment]
> Great work @Olusesan ! And was nice meeting you in person in SF!

Thanks @melonora! And it was really nice hanging out with you! We should do it again soon.

**Comment by TimMonko** [comment]
Added Sesan to the citation file.

It looks like `torch` is not available for mac intel, but I think we are considering dropping mac-intel for CI. Also seems like `geff` is having some resolution difficulties, but I don't have brain power to look into it.

File: pyproject.toml [hunk 20]
```diff
@@ -177,7 +177,9 @@ gallery = [
     "nilearn",
     "xarray",
     "h5netcdf",
+    "tifffile[codecs]",
     "meshio",
+    "trackastra",
```
Comment (DragaDoncila) [inline pyproject.toml]:
```suggestion
    "trackastra>=0.4.2",
```

File: pyproject.toml [hunk 21]
```diff
@@ -269,7 +271,9 @@ gallery = [
...
```
Comment (DragaDoncila) [inline pyproject.toml]:
```suggestion
    "trackastra>=0.4.2",
```

**Review by DragaDoncila — COMMENTED** [review]
Looks like the `lightning` package may have been the issue - eventually leading to `geff`. I'm going to bump the min req for trackastra because since 0.4.2 they've split out dependencies and we may get away with not installing lightning at all (or at least it wouldn't conflict with `geff` hopefully)

**Comment by DragaDoncila** [comment]
The docs build failure (at least) is real. I'm taking a look at how we can fix it.

File: pyproject.toml [hunk 22]
```diff
@@ -177,7 +177,9 @@ gallery = [
     "nilearn",
     "xarray",
     "h5netcdf",
+    "tifffile[codecs]",
     "meshio",
+    "trackastra>=0.4.2",
```
Comment (Czaki) [inline pyproject.toml:183]:
```suggestion
    "trackastra>=0.4.2 ; python_version < '3.13' or platform_system != 'Darwin' or platform_machine != 'x86_64'",
```

To still allow use of pixi on docs (torch drop support of macos intem for python 3.13+). 
The alternative will be to drop support of build docs on macos intels

Comment (TimMonko) [reply pyproject.toml:183]: My opinion is that we drop building docs on macos intel, it will simplify things going forwards as other projects are also not explicitly supporting such machines any more.

Comment (Czaki) [reply pyproject.toml:183]: Would you open a PR to docs?

Comment (TimMonko) [reply pyproject.toml:183]: @Czaki can you explain how we know this is a macos intel issue with torch? I'm not sure I'm seeing it in the logs

**Comment by DragaDoncila** [comment]
@Czaki looks like we're still getting the recursion error, even though I confirmed that the correct docs commit is checked out :sob:

**Comment by Czaki** [comment]
Opened https://github.com/thebjorn/pydeps/pull/242

If maintainer accepts, it will also require a docs PR.

As a temporary solution, we might try to increase the recursion limit even more. But it will be better if you @DragaDoncila test it as previous limit works on my machine.

**Comment by DragaDoncila** [comment]
@Czaki I played with the recursion limit locally a long time ago and was never able to increase it sufficiently to get the build going. It looks like the pydeps maintainer [asked for an update](https://github.com/thebjorn/pydeps/pull/242#issuecomment-3783927860) to the PR so hopefully that means we can get that in :grimacing:

**Comment by DragaDoncila** [comment]
@Czaki looks like this is finally green - are you happy with it? I had to bump the torchvision dependency to get it to play nice with torch 2.10

File: examples/tracks_3d_with_graph.py [hunk 23]
```diff
@@ -48,7 +48,7 @@ def _trajectory(t, r, track_id):
 vertices = tracks[:, 1:]
 
 viewer = napari.Viewer(ndisplay=3)
-viewer.add_points(vertices, size=1, name='points', opacity=0.3)
+# viewer.add_points(vertices, size=1, name='points', opacity=0.3)
```
Comment (Czaki) [inline examples/tracks_3d_with_graph.py]: uncomment?

File: pyproject.toml [hunk 24]
```diff
@@ -178,7 +178,10 @@ gallery = [
     "nilearn",
     "xarray",
     "h5netcdf[h5py]",
+    "tifffile[codecs]",
     "meshio",
+    # can simplify to just >=0.4.2 once we stop running macos intel docs
+    "trackastra>=0.4.2 ; python_version < '3.13' or platform_system != 'Darwin' or platform_machine != 'x86_64'",
```
Comment (Czaki) [inline pyproject.toml]: is this condition correct? python 3.12 will pass it on macos x86_64

**Review by Czaki — APPROVED** [review]
Looks ok. Two questions left

File: examples/tracks_3d_with_graph.py (hunk 23 above)
Comment (DragaDoncila) [reply examples/tracks_3d_with_graph.py]: Thanks good catch, this example shouldn't change. I'll push update.

File: pyproject.toml (hunk 24 above)
Comment (DragaDoncila) [reply pyproject.toml]: @Czaki idk lol you added this line!

Comment (Czaki) [reply pyproject.toml]: Ah. Its no longer needed with https://github.com/napari/docs/pull/913 merged

**Comment by DragaDoncila** [comment]
Woohoo this is finally in @Olusesan :tada: :heart_eyes:! Thank you for all your work and patience on this :pray:

**Comment by Olusesan** [comment]
I didn't expect this to break this many things and many thanks to you and the core dev team for fixing it. It's really nice to see how this came together and I hope it helps the community have a more enriching experience with Napari.
